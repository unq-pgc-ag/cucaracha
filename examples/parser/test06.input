fun main() {
  a := 10
  b := 11
  c := 12

  x := a <= b
  x := c >= b
  x := a < b
  x := b > c
  x := a == b
  x := b != a

  x := a < b and b < c
  x := a < b or b < c
  x := not a < b
  x := a < b and b < c and c < a
  x := a < b or b < c or c < a
  x := a < b and b < c or c < a
  x := a < b or b < c and c < a
  x := not a < b or not b < c
  x := not not a < b or b < c
  x := a < b or not not b < c
  x := not (a < b and b < c)
  x := not (a < b and b < c) or not (b < c and a < b)
  x := not (a < b and b < c) or not (b < c and a < b)
}
