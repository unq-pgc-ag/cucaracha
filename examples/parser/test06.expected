(Program
  (Function
    main
    Unit
    (Block
      (StmtAssign
        a
        (ExprConstNum
          10
        )
      )
      (StmtAssign
        b
        (ExprConstNum
          11
        )
      )
      (StmtAssign
        c
        (ExprConstNum
          12
        )
      )
      (StmtAssign
        x
        (ExprLe
          (ExprVar
            a
          )
          (ExprVar
            b
          )
        )
      )
      (StmtAssign
        x
        (ExprGe
          (ExprVar
            c
          )
          (ExprVar
            b
          )
        )
      )
      (StmtAssign
        x
        (ExprLt
          (ExprVar
            a
          )
          (ExprVar
            b
          )
        )
      )
      (StmtAssign
        x
        (ExprGt
          (ExprVar
            b
          )
          (ExprVar
            c
          )
        )
      )
      (StmtAssign
        x
        (ExprEq
          (ExprVar
            a
          )
          (ExprVar
            b
          )
        )
      )
      (StmtAssign
        x
        (ExprNe
          (ExprVar
            b
          )
          (ExprVar
            a
          )
        )
      )
      (StmtAssign
        x
        (ExprAnd
          (ExprLt
            (ExprVar
              a
            )
            (ExprVar
              b
            )
          )
          (ExprLt
            (ExprVar
              b
            )
            (ExprVar
              c
            )
          )
        )
      )
      (StmtAssign
        x
        (ExprOr
          (ExprLt
            (ExprVar
              a
            )
            (ExprVar
              b
            )
          )
          (ExprLt
            (ExprVar
              b
            )
            (ExprVar
              c
            )
          )
        )
      )
      (StmtAssign
        x
        (ExprNot
          (ExprLt
            (ExprVar
              a
            )
            (ExprVar
              b
            )
          )
        )
      )
      (StmtAssign
        x
        (ExprAnd
          (ExprAnd
            (ExprLt
              (ExprVar
                a
              )
              (ExprVar
                b
              )
            )
            (ExprLt
              (ExprVar
                b
              )
              (ExprVar
                c
              )
            )
          )
          (ExprLt
            (ExprVar
              c
            )
            (ExprVar
              a
            )
          )
        )
      )
      (StmtAssign
        x
        (ExprOr
          (ExprOr
            (ExprLt
              (ExprVar
                a
              )
              (ExprVar
                b
              )
            )
            (ExprLt
              (ExprVar
                b
              )
              (ExprVar
                c
              )
            )
          )
          (ExprLt
            (ExprVar
              c
            )
            (ExprVar
              a
            )
          )
        )
      )
      (StmtAssign
        x
        (ExprOr
          (ExprAnd
            (ExprLt
              (ExprVar
                a
              )
              (ExprVar
                b
              )
            )
            (ExprLt
              (ExprVar
                b
              )
              (ExprVar
                c
              )
            )
          )
          (ExprLt
            (ExprVar
              c
            )
            (ExprVar
              a
            )
          )
        )
      )
      (StmtAssign
        x
        (ExprAnd
          (ExprOr
            (ExprLt
              (ExprVar
                a
              )
              (ExprVar
                b
              )
            )
            (ExprLt
              (ExprVar
                b
              )
              (ExprVar
                c
              )
            )
          )
          (ExprLt
            (ExprVar
              c
            )
            (ExprVar
              a
            )
          )
        )
      )
      (StmtAssign
        x
        (ExprOr
          (ExprNot
            (ExprLt
              (ExprVar
                a
              )
              (ExprVar
                b
              )
            )
          )
          (ExprNot
            (ExprLt
              (ExprVar
                b
              )
              (ExprVar
                c
              )
            )
          )
        )
      )
      (StmtAssign
        x
        (ExprOr
          (ExprNot
            (ExprNot
              (ExprLt
                (ExprVar
                  a
                )
                (ExprVar
                  b
                )
              )
            )
          )
          (ExprLt
            (ExprVar
              b
            )
            (ExprVar
              c
            )
          )
        )
      )
      (StmtAssign
        x
        (ExprOr
          (ExprLt
            (ExprVar
              a
            )
            (ExprVar
              b
            )
          )
          (ExprNot
            (ExprNot
              (ExprLt
                (ExprVar
                  b
                )
                (ExprVar
                  c
                )
              )
            )
          )
        )
      )
      (StmtAssign
        x
        (ExprNot
          (ExprAnd
            (ExprLt
              (ExprVar
                a
              )
              (ExprVar
                b
              )
            )
            (ExprLt
              (ExprVar
                b
              )
              (ExprVar
                c
              )
            )
          )
        )
      )
      (StmtAssign
        x
        (ExprOr
          (ExprNot
            (ExprAnd
              (ExprLt
                (ExprVar
                  a
                )
                (ExprVar
                  b
                )
              )
              (ExprLt
                (ExprVar
                  b
                )
                (ExprVar
                  c
                )
              )
            )
          )
          (ExprNot
            (ExprAnd
              (ExprLt
                (ExprVar
                  b
                )
                (ExprVar
                  c
                )
              )
              (ExprLt
                (ExprVar
                  a
                )
                (ExprVar
                  b
                )
              )
            )
          )
        )
      )
      (StmtAssign
        x
        (ExprOr
          (ExprNot
            (ExprAnd
              (ExprLt
                (ExprVar
                  a
                )
                (ExprVar
                  b
                )
              )
              (ExprLt
                (ExprVar
                  b
                )
                (ExprVar
                  c
                )
              )
            )
          )
          (ExprNot
            (ExprAnd
              (ExprLt
                (ExprVar
                  b
                )
                (ExprVar
                  c
                )
              )
              (ExprLt
                (ExprVar
                  a
                )
                (ExprVar
                  b
                )
              )
            )
          )
        )
      )
    )
  )
)
